server/graphql/loaders/expenses.ts

```diff
@@ -1,12 +1,43 @@
 import DataLoader from 'dataloader';
 
 import ACTIVITY from '../../constants/activities';
-import models, { Op } from '../../models';
+import models, { Op, sequelize } from '../../models';
 import { ExpenseAttachedFile } from '../../models/ExpenseAttachedFile';
 import { ExpenseItem } from '../../models/ExpenseItem';
+import { LEGAL_DOCUMENT_TYPE } from '../../models/LegalDocument';
 
 import { sortResultsArray } from './helpers';
 
+const THRESHOLD = 600e2;
+const {
+  requestStatus: { RECEIVED },
+} = models.LegalDocument;
+
+
+const userTaxFormRequiredBeforePaymentQuery = `
+  SELECT 
+    e."UserId" "userId", 
+    MAX(er.id) as "expenseId",
+    MAX(ld."requestStatus") as "legalDocRequestStatus",
+    MAX(d."documentType") as "requiredDocument",
+    SUM (e."amount") AS total
+  FROM "Expenses" e
+  INNER JOIN "Expenses" er ON er.id IN (:expenseIds)
+  INNER JOIN "Collectives" c ON c.id = e."CollectiveId"
+  LEFT JOIN "RequiredLegalDocuments" d ON d."HostCollectiveId" = c."HostCollectiveId"
+                                    AND d."documentType" = 'US_TAX_FORM'
+  LEFT JOIN "LegalDocuments" ld ON ld."CollectiveId" = e."FromCollectiveId"
+                                AND ld.year = date_part('year', e."incurredAt")
+                                AND ld."documentType" = 'US_TAX_FORM'
+  WHERE e."UserId" = er."UserId"
+  AND e.status IN ('PENDING', 'APPROVED', 'PAID', 'PROCESSING', 'SCHEDULED_FOR_PAYMENT')
+  AND e.type NOT IN ('RECEIPT')
+  AND e."deletedAt" IS NULL
+  AND e."incurredAt" BETWEEN date_trunc('year', e."incurredAt") AND (date_trunc('year', e."incurredAt") + interval '1 year')
+  GROUP BY e."UserId"
+`;
```

Comment:
We're inverting the join condition and the filter condition here

```suggestion
  SELECT 
    e."UserId" "userId", 
    MAX(er.id) as "expenseId",
    MAX(ld."requestStatus") as "legalDocRequestStatus",
    MAX(d."documentType") as "requiredDocument",
    SUM (e."amount") AS total
  FROM "Expenses" e
  INNER JOIN "Expenses" er ON e."UserId" = er."UserId"
  INNER JOIN "Collectives" c ON c.id = e."CollectiveId"
  LEFT JOIN "RequiredLegalDocuments" d ON d."HostCollectiveId" = c."HostCollectiveId"
                                    AND d."documentType" = 'US_TAX_FORM'
  LEFT JOIN "LegalDocuments" ld ON ld."CollectiveId" = e."FromCollectiveId"
                                AND ld.year = date_part('year', e."incurredAt")
                                AND ld."documentType" = 'US_TAX_FORM'
  WHERE er.id IN (:expenseIds)
  AND e.status IN ('PENDING', 'APPROVED', 'PAID', 'PROCESSING', 'SCHEDULED_FOR_PAYMENT')
  AND e.type NOT IN ('RECEIPT')
  AND e."deletedAt" IS NULL
  AND e."incurredAt" BETWEEN date_trunc('year', e."incurredAt") AND (date_trunc('year', e."incurredAt") + interval '1 year')
  GROUP BY e."UserId"
`;
```